.github/workflows/publish.yaml

```diff
@@ -47,12 +47,11 @@ jobs:
           # https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable
           REF_NAME: ${{ github.ref_name }}
         run: |
-          echo "NPM_PACKAGE=$(echo $REF_NAME | grep -oE '(clients/js|contracts|integrations/(hardhat|wagmi-v2|viem-v2))')" >> $GITHUB_OUTPUT
+          echo "NPM_PACKAGE=$(echo $REF_NAME | grep -oE '(clients/js|contracts|integrations/(ethers-v6|hardhat|wagmi-v2|viem-v2))')" >> $GITHUB_OUTPUT
+          echo "NPM_TAG=$(echo $REF_NAME | grep -oP '(?<=\-)(rc|next|alpha|beta)' || echo 'latest')" >> $GITHUB_OUTPUT
       - name: Publish ${{ github.ref_name }} to NPM
-        uses: JS-DevTools/npm-publish@v3
-        with:
-          # Build runs as a prepublish script
-          ignore-scripts: false
-          token: ${{ secrets.NPM_TOKEN }}
-          package: ${{ steps.extract-tag.outputs.NPM_PACKAGE }}
-          access: public
+        run: pnpm publish --access public --no-git-checks --tag ${{ steps.extract-tag.outputs.NPM_TAG }}
```

Comment:
steps.extract-tag.outputs.NPM_TAG can be unsafe; move into env

Related https://github.com/oasisprotocol/sapphire-paratime/pull/346

Reply:
$REF_NAME will only ever be one of a number of known values, thus it is a trusted input.

Reply:
Additionally, which tags can be created are restricted, so even `github.ref_name` is a trusted input.

Originally the regexes were much stricter. I can change the pattern matchers, e.g. 

```
      - integrations/wagmi-v2/v[0-9]+.[0-9]+.[0-9]+
      - integrations/wagmi-v2/v[0-9]+.[0-9]+.[0-9]+-alpha.[0-9]+
      - integrations/wagmi-v2/v[0-9]+.[0-9]+.[0-9]+-beta.[0-9]+
      - integrations/wagmi-v2/v[0-9]+.[0-9]+.[0-9]+-next.[0-9]+
```

etc.

Reply:
Resolving as have moved it to environment variable